### DISCUSSION TITLE
Does .NET 6 support windows 98

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/runtime#79642] [Q&A] [answered]
Since .NET is aiming to become cross platform without relying on the OS, does the latest version  run on windows 98 which pretty much looks like windows 11

**Comment by GeraudFabien** [comment]
https://github.com/dotnet/core/blob/main/release-notes/6.0/supported-os.md
<html>
<body>


OS | Version | Architectures | Lifecycle
-- | -- | -- | --
Windows Client | 7 SP1(*), 8.1 | x64, x86 | Windows
Windows 10 Client | Version 1607+ | x64, x86, Arm64 | Windows
Windows 11 | Version 22000+ | x64, x86, Arm64 | Windows
Windows Server | 2012+ | x64, x86 | Windows Server
Windows Server Core | 2012+ | x64, x86 | Windows Server
Nano Server | Version 1809+ | x64 | Windows Server



> aiming to become cross platform without relying on the OS

Except if you are talking about WinUI 3 I don't know where this came from. And even WinUI3 need windows.

> latest version

.Net 6 is not the latest but 7 is.

> which pretty much looks like windows 11

...

**Comment by tannergooding** [comment, marked as answer]
> Since .NET is aiming to become cross platform

.NET has been cross platform for many years now.

> without relying on the OS

It is impossible to not rely on the OS. Cross platform applications rely on another runtime (such as the C or C++ runtime) and often a broader abstraction layer (.NET has the "PAL" or Platform Abstraction Layer) that helps to mask the differences between platforms and make it easier to write code that "runs anywhere".

In such a setup, new platforms are brought online by updating the abstraction layer to execute the necessary code for that target.

>  does the latest version run on windows 98 which pretty much looks like windows 11

While .NET is cross platform, that doesn't mean it runs "everywhere" and the same is true for most other languages/runtimes. Whether you look at Java, Rust, C/C++, Go, or even Javascript, most of these languages only run on versions of Windows, Linux, or MacOS that are still officially supported by their distributors. You'll be hard pressed to find modern versions that run on Windows Vista or Windows XP or in the case of Linux that run on Ubuntu 12 or MacOS 10.6.

Where these other languages/runtimes do have any support it is often not available via an official binary. Instead, it is most frequently community maintained and distributed. Such targets may correspondingly have a subpar experience or may not be able to support all modern features exposed.

In the case of Windows 98, it has been unsupported for many years now. It has a number of differences that would make it quite complex to target. Likewise, it does not trivially run on modern computers outside of an emulator.

**Reply by ghuntley** [reply]
Someone buy tanner a beer.

**Reply by nadjibus** [reply]
How about Windows 98 SE? 🤥

**Reply by john-holden-1** [reply]
Technically a "strictly conforming C/C++ program" doesn't rely on any OS or runtime and can target "freestanding environments" which doesn't have either.  Although they do still require the target CPU architecture to be Turing Complete 😃

**Comment by ForNeVeR** [comment]
There are some ways to run C# programs on older operating systems. [See this post](https://www.hanselman.com/blog/net-everywhere-apparently-also-means-windows-311-and-dos) for an example of running C# on Windows 3.11.

This is not easy at all, and requires some very specific knowledge. But doable provided you have enough time.

**Comment by GF-Huang** [comment]
Crazy idea, why not Win95 😆

**Comment by SuperJMN** [comment]
We need this. At least, for Windows 98 SE.
